src/backend/core/api/viewsets.py

```diff
@@ -696,18 +717,41 @@ def trashbin(self, request, *args, **kwargs):
 
         The selected items are those deleted within the cutoff period defined in the
         settings (see TRASHBIN_CUTOFF_DAYS), before they are considered permanently deleted.
+
+        Optimized version that uses EXISTS instead of expensive subqueries to check
+        owner access on items or their ancestors.
         """
-        queryset = self.queryset.select_related("creator").filter(
-            deleted_at__isnull=False,
-            deleted_at__gte=models.get_trashbin_cutoff(),
+        user = request.user
+
+        # Build the EXISTS subquery to check if user has owner access
+        # to the item or any of its ancestors
+        owner_access_exists = models.ItemAccess.objects.filter(
+            db.Q(user=user) | db.Q(team__in=user.teams),
+            role=models.RoleChoices.OWNER,
+            item__path__ancestors=db.OuterRef("path"),
         )
-        queryset = self.annotate_user_roles(queryset)
-        queryset = queryset.filter(user_roles__contains=[models.RoleChoices.OWNER])
```

Comment:
My experience in docs is that it is much faster (and simpler!) to do it in 2 queries:
```python
owned_documents_ids = models.ItemAccess.objects.filter(
    db.Q(user=user) | db.Q(team__in=user.teams),
    role=models.RoleChoices.OWNER,
    item__path__ancestors=db.OuterRef("path"),
).values_list("document_id", flat=True)

queryset = (
    self.queryset.select_related("creator")
    .filter(
        deleted_at__gte=models.get_trashbin_cutoff(),
    )
    .filter(id__in=owned_document_ids)
)
```
You just need to be sure that the list of owned documents ids fits in RAM but in this case definitely even if it is a few millions... and you'll just keep it a few nanoseconds.